src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,99 @@
+package duke.util;
+
+import duke.exception.DukeException;
+
+import java.util.Scanner;
+
+public class Parser {
+    private final Scanner sc;
+    private final List list;
+
+    public Parser(List list) {
+        this.sc = new Scanner(System.in);
+        this.list = list;
+    }
+
+    // extract int from done and delete prompts
+    public static int extractInt(String input) throws DukeException {
+        String[] parts = input.split(" ");//split along the whitespace to get the integer
+        if (parts.length <= 1) {// checking for incomplete prompts
+            throw new DukeException();
+        }
+        String numStr = parts[1];
+        return Integer.valueOf(numStr);
+    }
+
+    public void execute() {
```

Comment:
This method / while loop might be too long. At [this code quality link](https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#structure-code-logically), they stated the following:
> Be wary when a method is longer than the computer screen, and take corrective action when it goes
> beyond 30 LOC (lines of code). The bigger the haystack, the harder it is to find a needle.

This is a little longer than 30 LOC 😅